### ISSUE TITLE
[Misc] Storage Blocks

### ISSUE THREAD

**Issue by natrow** [natrow/tfc_minecolonies#11]
Minecolonies shelves <-> TFC balancing
 - Currently stores any item size (including tools)
 - More inventory space than TFC chests

TFC blocks to test/fix minecolonies support with:
 - Tool racks
 - Barrels
 - Chests
 - Bookshelves
 - Large vessels

**Comment by Chumpy83** [comment]
Villagers don’t seem to use storage outside of schematics, so I made a builder with only TFC storage blocks. None of them work presently. It looks like TFC storage is implemented differently than the MineColonies rack - seems they custom defined the rack to be minimize lag. There’s also a list of recognized containers (ModContainers.class) I’m sure things need added to. I’m trying, rather in vain at the moment, to copy MC’s code for the rack (AbstractTileEntityRack.class) and make it a more generic container class, not sure how much progress I’ll make. Figured from there you can mix it in with whatever mod added storage might exist?

Def feel like I’m punching above my weight class here - gonna watch a good chunk of Java tutorial/refresher videos and see if I go a little less cross-eyed in another couple of weeks maybe? lol

**Comment by natrow** [comment]
Yeah, for all of the issues labeled "question," I haven't done much (or any) research into them. Ideally, there would be a way to get minecolonies workers to be able to use any of the TFC storage blocks. If you can find what class(es) minecolonies uses to actually get their workers to interact with storage containers then you could mixin to that... I recall them being able to interact with vanilla chests, hut blocks, and racks. You could start by looking for uses of that AbstractTileEntityRack within any EntityAIWorker type class
